examples/simple/src/Layout.tsx

```diff
@@ -1,21 +1,29 @@
 import * as React from 'react';
 import { forwardRef, memo } from 'react';
 import { ReactQueryDevtools } from 'react-query/devtools';
-import { Layout, AppBar, UserMenu, useLocaleState } from 'react-admin';
+import {
+    AppBar,
+    Layout,
+    Logout,
+    UserMenu,
+    useLocaleState,
+    useUserMenu,
+} from 'react-admin';
 import { MenuItem, MenuItemProps, ListItemIcon } from '@mui/material';
 import Language from '@mui/icons-material/Language';
 
 const SwitchLanguage = forwardRef<HTMLLIElement, MenuItemProps>(
     (props, ref) => {
         const [locale, setLocale] = useLocaleState();
+        const { onClose } = useUserMenu();
 
         return (
             <MenuItem
                 ref={ref}
                 sx={{ color: 'text.secondary' }}
                 onClick={event => {
                     setLocale(locale === 'en' ? 'fr' : 'en');
-                    props.onClick(event);
+                    onClose();
```

Comment:
we need to document this new hook and update the doc for custom menus

Reply:
Yes, that's why I tagged the PR as `WIP` and added a task item in the PR description todo ;-)